nixos/modules/services/networking/fastnetmon-advanced.nix

```diff
@@ -0,0 +1,204 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.fastnetmon-advanced;
+  settingsFormat = pkgs.formats.yaml { };
+
+  # obtain the default configs by starting up ferretdb and fcli in a VM
+  default_configs = pkgs.vmTools.runInLinuxVM (pkgs.runCommand "default-configs" {} ''
+    mkdir ferretdb
+    FERRETDB_TELEMETRY="disable" FERRETDB_HANDLER="sqlite" FERRETDB_STATE_DIR="$PWD/ferretdb" FERRETDB_SQLITE_URL="file:$PWD/ferretdb/" ${pkgs.ferretdb}/bin/ferretdb &
+
+    mkdir -p /etc/fastnetmon/keychain
+    cat << EOF > /etc/fastnetmon/fastnetmon.conf
+    ${builtins.toJSON {
+      mongodb_username = "";
+    }}
+    EOF
+    ${pkgs.fastnetmon-advanced}/bin/fcli create_configuration
+    ${pkgs.fastnetmon-advanced}/bin/fcli set bgp default
+    ${pkgs.fastnetmon-advanced}/bin/fcli export_configuration backup.tar
+    tar -C $out --no-same-owner -xvf backup.tar
+  '');
+
+  # merge the user configs into the default configs
+  config_tar = pkgs.runCommand "fastnetmon-config.tar" {
+    nativeBuildInputs = with pkgs; [ jq ];
+  } ''
+    jq -s add ${default_configs}/main.json ${pkgs.writeText (baseNameOf "main-add.json") (builtins.toJSON cfg.settings)} > main.json
```

Comment:
Why are you using baseNameOf here? 
doing `builtins.baseNameOf "main-add.json"` in a nix repl just returns the same value

Reply:
Good question 🤔 
Probably historically grown ™️ 
removed it